MyCryptoHQ/MyCrypto
# Electron TREZOR Support

Builds off of #1836, finishes the other half of #1621.

### Description

Implements the TREZOR side of hardware wallet integration in Electron using the enclave library introduced in #1621. Similar to that PR, there's an unfortunate amount of code duplication between the two wallet implementations. Any feedback on how to better unite those would be appreciated.

See "Steps to Test" for note about pending PR. I have personally tested that this all works.

### Changes

* Implemented TREZOR on Enclave side
* Re-enabled TREZOR wallet in Electron environment
* Typed up all TREZOR libraries (including the vendor'd `TrezorConnect`.)

### Steps to Test

**NOTE:** This is still pending on https://github.com/trezor/trezor-link/pull/9. If you wish to test this PR locally, you will have to manually make the changes proposed their to `node_modules/trezor-link/lib/http.js`.

1. Start electron app
2. Unlock a TREZOR wallet address
3. Display the address on your device
4. Sign a transaction, confirm it looks good
5. Repeat those steps in the web app
